### DISCUSSION TITLE
Is it possible to get the triangle mesh data out of ospray::cpp::World

### DISCUSSION THREAD

**Discussion by fatecasino** [RenderKit/ospray#578] [General]
I wanted to ask if it is possible to do the reverse process of ospTutorial.cpp and get the actual triangle mesh data (std::vector<vec3f> vertex) out of the World object. Any code example would be more than helpful!

**Comment by johguenther** [comment]
This is not possible. OSPRay as a rendering library has a "write only" API. For some devices / backends like GPU or MPI the data may even not be available anymore in the local host's memory.
If you want to operate on scene data it is better one level above, e.g. at a scenegraph lib.

**Reply by fatecasino** [reply]
thanks for the quick response, indeed it's a write only API, I was trying in vain to find a getter method :)
